### DISCUSSION TITLE
Cropping to content in large images

### DISCUSSION THREAD

**Discussion by nixin72** [ImageMagick/ImageMagick#5058] [Help] [answered]
I have a bunch of files that need cropping, but I need to crop them to the actual content of the image, not just crop them to a predetermined size/placement/aspect ratio/etc. Was I was going to do was loop over all my images and do `convert path/to/img txt:`, find the left/top/right/bottom-most non-transparent pixel, and then crop to those as my bounds. However, I didn't realize my images were super high resolution and I'm dealing with 3000x4000 pixels in some cases, which would make this really slow. Is there a way to either crop-to-content more easily? Or if not, some good way to get these boundary non-transparent pixels without looping over all of them myself so that I can crop to the content manually using these bounds? 

For example, say I have this, where the `0`s are fully transparent white pixels (0,0,0,0), and the spaces are anything else.
```
00000000000000000000
000000000  000000000
0000000      0000000
00000          00000
0000000      0000000
000000000  000000000
00000000000000000000
```
Here, my boundaries would be (top, right, bottom, left): 1, 5, 14, 5, which I'd crop to those so that I just have
```
0000  0000
00      00
          
00      00
0000  0000
```
Is there an efficent way to do this?

**Comment by fmw42** [comment, marked as answer]
If your background is nearly one constant color, then you can use the automatic crop tool called -trim. See https://legacy.imagemagick.org/Usage/crop/#trim

If you need to crop all the images at the same crop bounding box, you can use the string format %@ to get those bounds and then crop all the images with the same command using -crop. See https://imagemagick.org/script/escape.php and https://legacy.imagemagick.org/Usage/crop/#crop

**Reply by nixin72** [reply]
Thank you! I found the crop, but since I needed to supply something to tell it how to crop, I thought I had to find my boundaries manually. Using `convert input.png -trim +repage output.png` works great though!
